### DISCUSSION TITLE
dependencies 是 array-table 时，变更无法触发 reactions 响应

### DISCUSSION THREAD

**Discussion by prgrmrwy** [alibaba/formily#1620] [Q&A 2.x]
问题描述：
其他字段通过 dependencies 关联 array-table 后，array-table 内容变化不会触发字段响应，看起来是 array-table 的引用没变化导致的？
预期：array-table 内容变更时可以触发字段响应
目的：想确认下是否属于 bug，或者有什么建议

源码：https://codesandbox.io/s/winter-sound-t1g57?file=/App.tsx:3852-3891

**Comment by janryWang** [comment]
array table内容变化是不会触发整体array数据变化的，你需要更加精确的依赖才行
```
input: {
      type: "string",
      title: "123",
      "x-decorator": "FormItem",
      "x-component": "Input",
      "x-reactions": {
        dependencies: ["array.0.a1"],
        fulfill: {
          schema: {
            description: "{{$deps[0]}}"
          },
          run: "test($self, $deps[0])"
        }
      }
    }
```

**Reply by prgrmrwy** [reply]
明白，主要是我的场景是：同一份数据两种输入形式，目前的做法是希望可以一份输入的时候，另一份可以实时转换
比如一个是 table 输入，另一个是 json 输入，当 table tab 输入时，希望能够直接转换到 json 格式

**Reply by janryWang** [reply]
那你就在依赖数据的地方深度clone array数据

**Reply by prgrmrwy** [reply]
请问有示例用法么，深度 clone 感觉也需要触发时机，感觉目前没啥好办法可以触发 clone

**Reply by boyrest** [reply]
@programmerwy 老哥，最后这个怎么深度clone的

**Reply by janryWang** [reply]
JSON.parse(JSON.stringify(value))

**Reply by prgrmrwy** [reply]
> @programmerwy 老哥，最后这个怎么深度clone的

绕过去的。。他们没提供触发时机，深度 clone 不是问题所在，最后通过 consumer 绕过去的，不过其他场景还是没法弄

**Reply by prgrmrwy** [reply]
开了个类似的新 disscuss: https://github.com/alibaba/formily/discussions/1922
